tests/test_core.py

```diff
@@ -1163,21 +1167,36 @@ def test_objcinstance_python_attribute(self):
         self.assertEqual(thing.python_object_1, range(2, 8))
         self.assertEqual(thing.python_object_2, type)
 
+        # Test deleting Python attribute.
+
+        del thing.python_object_1
+
+        with self.assertRaises(AttributeError):
+            thing.python_object_1
+
     def test_objcinstance_python_attribute_keep_alive(self):
         """Python attributes on an ObjCInstance are kept even if the object temporarily has no Python references."""
 
-        Example = ObjCClass("Example")
-        example = Example.alloc().init()
-        Thing = ObjCClass("Thing")
-        thing = Thing.alloc().init()
+        example = NSObject.alloc().init()
+        thing = NSObject.alloc().init()
+
+        # create NSAutoreleasePool to clean up any leftover Obj-C instances
+        pool = NSAutoreleasePool.alloc().init()
```

Comment:
This isn't something that needs to be done in this PR, but ideally Rubicon should provide a proper API for autorelease pools, probably as a context manager (`with objc.autoreleasepool(): ...`).

In normal Objective-C code, `NSAutoreleasePool` is more or less superseded by `@autoreleasepool` blocks, and when using ARC, `@autoreleasepool` is the only option - see the warning in [the `NSAutoreleasePool` docs](https://developer.apple.com/documentation/foundation/nsautoreleasepool?language=objc). Internally, `@autoreleasepool` blocks are translated to calls to the Objective-C runtime functions `objc_autoreleasePoolPush` and `objc_autoreleasePoolPop`, which are undocumented and not declared in Apple's headers, but [seem to be widely known](https://stackoverflow.com/q/15014152). It's *probably* fine if we keep using `NSAutoreleasePool` for now - it's not officially deprecated, and we aren't affected by ARC restrictions. But for future-proofing it would be better to define our own high-level API for autorelease pools that uses the appropriate mechanism under the hood. (Again, not something that needs to be done in this PR.)

Reply:
Yes, an `autoreleasepool` context manager would be a good idea. A lot of code that uses rubicon-objc will likely also run a Cocoa event loop which will create and drain pools for each cycle but there of course will remain use cases for manually managing an autorelease pool.

I'm not so sure about using undocumented Objective-C runtime functions. Could this potentially lead to issues with app review if they are considered private? I have never gone through the review process myself...

Reply:
I've copied the discussion into a separate issue and replied to your question there: #210